src/main/scala/mimir/Database.scala

```diff
@@ -183,14 +183,13 @@ case class Database(backend: Backend)
 
 
   /**
-   * Make an educated guess about what the query's schema should be
-   * XXX: Only required because the TypeInference lens is a little punk birch that needs to
-   *      be converted into an adaptive schema (#193).
+   * Shortcut to get the schema of an operator.  The TypeInference and
+   *  SchemaMatching lenses have been killed and now can exist only as 
+   *  adaptive schemas so no more bestGuessSchema!  
    */
-  def bestGuessSchema(oper: Operator): Seq[(String, Type)] =
+  def schemaOf(oper: Operator): Seq[(String, Type)] =
```

Comment:
Throw this function away, imo.  Route everything through typechecker.
`
db.typechecker.schemaOf
` 
makes it clearer what portion of the system is being invoked.